tests/commands/test_history.py

```diff
@@ -1,137 +1,175 @@
 import subprocess
 from pathlib import Path
+
+from onyo.lib import Repo
 import pytest
 
 
 # NOTE: the output of `onyo history` is not tested for formatting or content, as
 #       the commands called by `onyo history` are user-configurable. Instead, it
 #       is tested for whether the output of the underlying command is unaltered.
-variants = [
-    'file',
-    'dir',
-    'dir/subfile-1',
-    's p a c e s/subfile-2',
-    's p a/c e s/sub file 3',
-]
-@pytest.mark.repo_files('file', 'dir/subfile-1', 's p a c e s/subfile-2', 's p a/c e s/sub file 3')
-@pytest.mark.parametrize('variant', variants)
-def test_history_noninteractive(repo, variant):
-    ret = subprocess.run(['onyo', 'history', '-I', variant],
+
+
+files = ['laptop_apple_macbookpro',
+         'lap top_ap ple_mac book pro',
+         'spe\"c_ialch_ar\'ac.teஞrs']
+
+directories = ['.',
+               's p a c e s',
+               'very/very/very/deep',
+               ]
+
+assets = [f"{d}/{f}.{i}" for f in files for i, d in enumerate(directories)]
+
+@pytest.mark.repo_files(*assets)
+@pytest.mark.parametrize('asset', assets)
+def test_history_noninteractive(repo: Repo, asset: str) -> None:
+    """
+    Test that the default `onyo history -I ASSET` command runs and print to
+    std.out, but not std.err.
+    """
+    ret = subprocess.run(['onyo', 'history', '-I', asset],
                          capture_output=True, text=True)
     assert ret.returncode == 0
     assert ret.stdout
     assert not ret.stderr
+    repo.fsck()
 
 
-variants = [
-    'does-not-exist',
-    'subdir/does-not-exist',
+no_assets = [
+    'does_not_exist.test',
+    'subdir/does_not_exist.test',
 ]
-@pytest.mark.repo_dirs('subdir')
-@pytest.mark.parametrize('variant', variants)
-def test_history_noninteractive_not_exist(repo, variant):
-    ret = subprocess.run(['onyo', 'history', '-I', variant],
+@pytest.mark.repo_dirs(*assets)
```

Comment:
flake8 does not appear to address: `PEP 8: E302 expected 2 blank lines, found 0` on the function definition, because the closing bracket of the variable assignment is on the line before it. Furthermore, while your approach works with the way pytest evaluates a test file and executes the tests, it can easily break if the test order is randomized, and the variable is reassigned later in the file.

Rather than creating a variable and passing it to `parametrize`, I suggest writing the contents of the variable in there, e.g.,

```
@pytest.mark.parametrize('variant', ['does_not_exist.test', 'subdir/does_not_exist.test'])
```

Reply:
You are right, the definitions of variants should happen there, it just works for the static ones that are the same for all situations anyways, without being ever modified. Thanks! :)